File: fixtures/java/Dog.kt
package com.example.kotlin

/**
 * 狗类 - Kotlin 继承
 */
class Dog(name: String, age: Int, private val breed: String) : Animal(name, age) {
    
    override fun makeSound() {
        println("$name says: Woof! Woof!")
    }
    
    fun fetch() {
        println("$name is fetching the ball")
    }
    
    fun getBreed(): String {
        return breed
    }
}

File: fixtures/java/Dog.cs
using System;

namespace Example.Models
{
    /// <summary>
    /// 狗类 - C# 继承
    /// </summary>
    public class Dog : Animal
    {
        private string breed;
        
        public Dog(string name, int age, string breed) : base(name, age)
        {
            this.breed = breed;
        }
        
        public override void MakeSound()
        {
            Console.WriteLine($"{name} says: Woof! Woof!");
        }
        
        public void Fetch()
        {
            Console.WriteLine($"{name} is fetching the ball");
        }
        
        public string GetBreed() => breed;
    }
}
